helper/schema/resource_diff.go

```diff
@@ -378,6 +378,25 @@ func (d *ResourceDiff) GetOk(key string) (interface{}, bool) {
 	return r.Value, exists
 }
 
+// GetOkExists functions the same way as GetOkExists within ResourceData, but
+// it also checks the new diff levels to provide data consistent with the
+// current state of the customized diff.
```

Comment:
Likewise this docstring seems to describe how it achieves its goal rather than what it actually does. In this particular case I'm honestly not really sure I fully understand the implication of this in relation to `GetOk` even after reading the implementation, since I'm rusty on the internal behavior here, so perhaps it would be helpful to write this docstring to focus on how it is different from `ResourceDiff.GetOk` rather than how it is similar/different from `ResourceData.GetOkExists`.